app/models/garden.rb

```diff
@@ -7,4 +7,15 @@ class Garden < ApplicationRecord
   has_many :sensors
   has_many :garden_plants
   has_many :plants, through: :garden_plants
+
+  def add_plant(plant)
+    GardenPlant.create!(garden: self, plant: plant, planted_date: Date.today)
+  end
+
+  def create_and_add_plant(plant_params)
+    plant = Plant.new(plant_params)
+    if plant.save
+      GardenPlant.create!(garden: self, plant: plant, planted_date: Date.today)
+    end
+  end
```

Comment:
If front end wants to let the user specify which date they planted the plant, we will need them to send with params.